internal/mcp-router/response_test.go

```diff
@@ -0,0 +1,85 @@
+package mcprouter
+
+import (
+	"context"
+	"log/slog"
+	"os"
+	"testing"
+
+	basepb "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	eppb "github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3"
+	"github.com/kagenti/mcp-gateway/internal/broker"
+	"github.com/kagenti/mcp-gateway/internal/cache"
+	"github.com/kagenti/mcp-gateway/internal/config"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRouter404Detection(t *testing.T) {
+	logger := slog.New(slog.NewTextHandler(os.Stdout, nil))
+
+	// Create test session cache
+	sessionCache := cache.New(func(_ context.Context, _ string, _ string, _ string) (string, error) {
+		return "test-mcp-session-123", nil
+	})
+
+	server := &ExtProcServer{
+		RoutingConfig: &config.MCPServersConfig{
+			Servers: []*config.MCPServer{
+				{
+					Name:     "test-server",
+					URL:      "http://test-server:9090/mcp",
+					Hostname: "test-server",
+					Enabled:  true,
+				},
+			},
+		},
+		Broker:       broker.NewBroker(logger),
+		SessionCache: sessionCache,
+	}
+
+	// Create response headers with HTTP 404 status
+	responseHeaders := &eppb.HttpHeaders{
+		Headers: &basepb.HeaderMap{
+			Headers: []*basepb.HeaderValue{
+				{
+					Key:      ":status",
+					RawValue: []byte("404"),
+				},
+				{
+					Key:      "mcp-session-id",
+					RawValue: []byte("test-mcp-session-123"),
+				},
+			},
+		},
+	}
+
+	// Test that 404 detection works
+	resp, err := server.HandleResponseHeaders(responseHeaders)
+	require.NoError(t, err)
+	require.Len(t, resp, 1)
+}
+
+func TestInvalidateByMCPSessionID(t *testing.T) {
```

Comment:
Not sure I am understanding this test well. Seems like whether the session exists or not we would get the same result, so not really testing that the session exists or was created ?